lib/host_delete.py

```diff
@@ -1,41 +1,53 @@
+from kafka.errors import KafkaError
 from sqlalchemy.orm.base import instance_state
 
+from app.logging import get_logger
 from app.models import Host
 from app.queue.events import build_event
 from app.queue.events import EventType
 from app.queue.events import message_headers
+from lib.db import session_guard
+from lib.host_kafka import kafka_available
 from lib.metrics import delete_host_count
 from lib.metrics import delete_host_processing_time
 
 __all__ = ("delete_hosts",)
+logger = get_logger(__name__)
 
 
 def delete_hosts(select_query, event_producer, chunk_size, interrupt=lambda: False):
-    while select_query.count():
-        for host in select_query.limit(chunk_size):
-            host_id = host.id
-            with delete_host_processing_time.time():
-                _delete_host(select_query.session, host)
+    with session_guard(select_query.session):
+        while select_query.count():
+            for host in select_query.limit(chunk_size):
+                host_id = host.id
+                with delete_host_processing_time.time():
+                    host_deleted = _delete_host(select_query.session, event_producer, host)
```

Comment:
The KafkaError should be handled here to stop deletion if Kafka stops being available during a bulk deletion 

```suggestion
               try:
                   host_deleted = _delete_host(select_query.session, event_producer, host)
               except KafkaError:
                   logger.error("KafkaError: Stopping bulk deletion")
                   raise
```

Reply:
The KafkaError is handled by the caller in `api.host`